Packs/IntSight/Integrations/Rapid7ThreatCommand/Rapid7ThreatCommand.py

```diff
@@ -0,0 +1,5325 @@
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    validate_create_alert(args=args)
+    response = client.create_alert(
+        title=args["title"],
+        description=args["description"],
+        type_=args.get("type"),
+        sub_type=args.get("sub_type"),
+        severity=args["severity"],
+        source_type=args["source_type"],
+        source_network_type=args["source_network_type"],
+        source_date=args.get("source_date"),
+        found_date=args.get("found_date"),
+        image_entry_ids=argToList(args.get("image_entry_ids")),
+        scenario=args.get("scenario"),
+        source_url=args.get("source_url"),
+    )
+    outputs = {"id": response.content.decode()}
+    return command_result_generate(
+        readable_message=ReadableOutputs.ALERT_CREATE.value,
+        outputs=outputs,
+        headers=["id"],
+        prefix="Alert",
+        key_field="id",
+        raw_response=outputs,
+    )
+
+
+def close_alert_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    """
+    Close alert.
+
+    Args:
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    validate_close_alert(args)
+
+    alert_id = args["alert_id"]
+
+    client.close_alert(
+        alert_id=alert_id,
+        reason=args["reason"],
+        comment=args.get("comment"),
+        is_hidden=argToBoolean(args["is_hidden"]),
+        rate=arg_to_number(args.get("rate")),
+    )
+
+    return CommandResults(
+        readable_output=ReadableOutputs.ALERT_CLOSE.value.format(alert_id)
+    )
+
+
+def update_alert_severity_command(
+    client: Client, args: dict[str, Any]
+) -> CommandResults:
+    """
+    Update the alert severity.
+
+    Args:
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    validate_alert_ioc_severity(args)
+
+    alert_id = args["alert_id"]
+    severity = args["severity"]
+
+    client.update_alert_severity(alert_id=alert_id, severity=severity)
+
+    return CommandResults(
+        readable_output=ReadableOutputs.ALERT_SEVERITY.value.format(alert_id, severity)
+    )
+
+
+def assign_alert_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    """
+    Assign alert to user.
+
+    Args:
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    validate_assign_alert(args)
+    alert_id = args["alert_id"]
+    user_id = args["user_id"]
```

Comment:
assuming only one analyst is assigned to an alert, let's add it to the context, and when it's unassigned, set to null